pkg/streamingpromql/types/fpoint_ring_buffer.go

```diff
@@ -183,60 +166,100 @@ func (b *FPointRingBuffer) Close() {
 	b.points = nil
 }
 
-// First returns the first point in this buffer.
-// It panics if the buffer is empty.
-func (b *FPointRingBuffer) First() promql.FPoint {
-	if b.size == 0 {
-		panic("Can't get first element of empty buffer")
+type FPointRingBufferView struct {
+	buffer *FPointRingBuffer
+	size   int
+}
+
+// UnsafePoints returns slices of the points in this buffer view.
+// Either or both slice could be empty.
+// Callers must not modify the values in the returned slices or return them to a pool.
```

Comment:
(nit, I know this existed previously) could be read as "callers must return them to a pool".

```suggestion
// Callers must not modify the values in the returned slices nor return them to a pool.
```